profiler/options.go

```diff
@@ -246,10 +246,10 @@ func defaultConfig() (*config, error) {
 		WithUploadTimeout(d)(&c)
 	}
 	if v := os.Getenv("DD_API_KEY"); v != "" {
-		WithAPIKey(v)(&c)
+		c.apiKey = v
 	}
 	if internal.BoolEnv("DD_PROFILING_AGENTLESS", false) {
-		WithAgentlessUpload()(&c)
+		c.agentless = true
 	}
```

Comment:
NIT: Isn't this the same as `c.agentless = internal.BoolEnv("DD_PROFILING_AGENTLESS", false)`?